File: client/src/boundary/interactor/ShipD2BConverter.ts
import { Ship } from "domain/index";
import { BoundaryShip } from "boundary/model/index";
import { Converter } from "utils/Converter";

export class ShipD2BConverter extends Converter<Ship, BoundaryShip> {
  convert(ship: Ship): BoundaryShip {
    return new BoundaryShip(
      ship.length,
      ship.coordinates,
      ship.hits,
      ship.destroyed
    );
  }
}

File: server/battleships/use_cases/implementation/src/main/java/eu/optas/use_cases/implementation/ShipD2BConverter.java
package eu.optas.use_cases.implementation;

import eu.optas.domain.Ship;
import eu.optas.use_cases.api.BoundaryShip;
import eu.optas.utils.Converter;

public class ShipD2BConverter extends Converter<Ship, BoundaryShip> {
    @Override
    public BoundaryShip convert(Ship ship) {
        return new BoundaryShip(ship.getLength(), ship.getCoordinates(), ship.getHits(), ship.isDestroyed());
    }
}
